### DISCUSSION TITLE
File not opening if path different from currently opened file in neovide

### DISCUSSION THREAD

**Discussion by Praful** [neovide/neovide#2667] [General]
**Describe the bug**
I have neovide running as a service using:
```
neovide -- --listen localhost:12345 &
```
I open a file from the shell using:
```
nvim --server localhost:12345 --remote file1.txt
```
If I then change folder and try to open another file, a blank file with the name 
opens in neovide.

**To Reproduce**
1. Start neovide: `neovide -- --listen localhost:12345 &`
2. `cd test1` then open a file in that folder: `nvim --server localhost:12345 --remote file1.txt`
3. File `file1.txt` will open.
4. `cd test2` and open another file: `nvim --server localhost:12345 --remote file2.txt`
5. A blank file called `file2.txt` will open in neovide _not_ the actual file.
6. In neovide, if I type: `:pwd` when `file2.txt` is open, the current folder will be that of `file1.txt`.

**Expected behavior**
I expect neovide to open both files.

**Desktop**
 - OS: Linux Mint 21.2
 - Neovide Version: 0.13.1
 - Neovim Version: 0.10.0

**Additional context**
I'm new to nvim/neovide (I use gvim normally). This is almost certainly _not_ a bug but 
rather me doing something wrong. Please let me know how to open files in the same instance (but different buffers) of neovide.

The above reproduces the issue but I currently have this helper function (which has the same issue) in my `.zshrc` file to open a file using nvim/neovide eg using `nv file.txt`:

```
# open file in existing neovide instance and move focus to it
function nv() {
  PORT=12345
  if lsof -i :$PORT > /dev/null; then
    echo "Neovide is already running."
  else
    neovide -- --listen localhost:$PORT &
    # wait for neovide to start otherwise nvim can't see the server
    sleep 1
  fi

  nvim --server localhost:$PORT --remote $@

  # move focus to neovide
  xdotool search --onlyvisible --class neovide windowactivate
}
```

**Comment by fredizzimo** [comment]
If this is a bug, it should be reported to Neovim, since the command `nvim --server localhost:$PORT --remote $@` does not involve Neovide at all. And you will see the same behaviour in terminal Neovim as well. But I don't think it can be fixed in the general case, since the server might run on a different machine and wont be able to open local files.

I would change your script to something like `nvim --server localhost:$PORT --remote $(readlink -f $1)`. 

NOTE: We will probably add native Neovide support for this at some point, see 
* https://github.com/neovide/neovide/issues/1586

**Comment by Praful** [comment]
@fredizzimo thanks for replying. I'll raise this issue with nvim.

Thanks for your suggestion about using `readlink`, which I was unaware of. `man readlink` recommended using `realpath`. The amended function can take multiple files (courtesy of `realpath`) and send them to neovide:

```
# open file in existing neovide instance and move focus to it
function nv() {
  local port=12345
  if lsof -i :$port > /dev/null; then
    echo "Neovide is already running."
  else
    neovide -- --listen localhost:$port &
    # wait for neovide to start otherwise nvim can't see the server
    sleep 1
  fi

  nvim --server localhost:$port --remote $(realpath $@)
  xdotool search --onlyvisible --class neovide windowactivate
}
```

I'd suggest including something like the above in the FAQ because it took me longer than expected to work it out and it may help others.

What is the difference between starting a listener using `nvim` (which is in the link you provided) and starting it in `neovide` as I've done?

**Reply by fredizzimo** [reply]
The reason I suggested `readlink -f` is that there are severak different `realpath` available, and some of them might not deal with non-existing files the way you probably want. Or it might need an additional parameter, see https://unix.stackexchange.com/questions/136494/whats-the-difference-between-realpath-and-readlink-f

> What is the difference between starting a listener using nvim (which is in the link you provided) and starting it in neovide as I've done?

Probably nothing other than that in the link I gave, Neovide is started separately. I did not even know it was possible to do it the way you did, which I think both embeds it snd starts a listening server.

**Reply by Praful** [reply]
Thanks for the reference. I just checked and found out that I have the GNU versions of `realpath` and `readlink`. The link you provided says the GNU `realpath` is equivalent to `readlink -f` -- which is probably why it worked for me!

Both result in existing files opening correctly and non-existing files opening as empty files in Neovide, even when multiple files are provided. That's good enough for me. And now I know about the two utilities, I'll keep an eye on the behaviour and adjust accordingly.

I don't know where I got the `neovide --listen` syntax because `neovide --help` doesn't show it!
